akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/ActiveActiveShardingSpec.scala

```diff
@@ -0,0 +1,144 @@
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.MemberStatus
+import akka.cluster.sharding.typed.scaladsl.Entity
+import akka.cluster.typed.Cluster
+import akka.cluster.typed.Join
+import akka.persistence.testkit.PersistenceTestKitPlugin
+import akka.persistence.testkit.query.scaladsl.PersistenceTestKitReadJournal
+import akka.persistence.typed.ReplicaId
+import akka.persistence.typed.scaladsl.ActiveActiveEventSourcing
+import akka.persistence.typed.scaladsl.Effect
+import akka.persistence.typed.scaladsl.EventSourcedBehavior
+import akka.serialization.jackson.CborSerializable
+import com.typesafe.config.ConfigFactory
+import org.scalatest.wordspec.AnyWordSpecLike
+
+object ActiveActiveShardingSpec {
+  def config = ConfigFactory.parseString("""
+      akka.loglevel = DEBUG
+      akka.loggers = ["akka.testkit.SilenceAllTestEventListener"]
+      akka.actor.provider = "cluster"
+      # pretend we're a node in all dc:s
+      akka.cluster.roles = ["DC-A", "DC-B", "DC-C"]
+      akka.remote.classic.netty.tcp.port = 0
+      akka.remote.artery.canonical.port = 0""").withFallback(PersistenceTestKitPlugin.config)
+}
+
+class ActiveActiveShardingSpec
+    extends ScalaTestWithActorTestKit(ActiveActiveShardingSpec.config)
+    with AnyWordSpecLike
+    with LogCapturing {
+
+  object MyActiveActiveStringSet {
+    trait Command extends CborSerializable
+    case class Add(text: String) extends Command
+    case class GetTexts(replyTo: ActorRef[Texts]) extends Command
+    case class Texts(texts: Set[String]) extends CborSerializable
+
+    def apply(entityId: String, replicaId: ReplicaId, allReplicas: Set[ReplicaId]): Behavior[Command] =
+      ActiveActiveEventSourcing.withSharedJournal(
+        entityId,
+        replicaId,
+        allReplicas,
+        PersistenceTestKitReadJournal.Identifier) { aaContext =>
+        EventSourcedBehavior[Command, String, Set[String]](
+          aaContext.persistenceId,
+          Set.empty[String],
+          (state, command) =>
+            command match {
+              case Add(text) =>
+                Effect.persist(text)
+              case GetTexts(replyTo) =>
+                replyTo ! Texts(state)
+                Effect.none
+            },
+          (state, event) => state + event).withJournalPluginId(PersistenceTestKitPlugin.PluginId)
+      }
+  }
+
+  object ProxyActor {
+    sealed trait Command
+    case class ForwardToRandom(entityId: String, msg: MyActiveActiveStringSet.Command) extends Command
+    case class ForwardToAll(entityId: String, msg: MyActiveActiveStringSet.Command) extends Command
+
+    def apply(): Behavior[Command] = Behaviors.setup { context =>
+      // #bootstrap
+      val aaShardingSettings =
+        ActiveActiveShardingSettings[
+          MyActiveActiveStringSet.Command,
+          ShardingEnvelope[MyActiveActiveStringSet.Command]](
```

Comment:
Do we need a second type param here? Is that not so that `E` will always be `ShardingEnvelope[M]`?

It will be good if can hide this from the user.

Reply:
I also find it annoying to have `ActiveActiveShardingSettings` and `ActiveActiveShardingReplicaSettings`. 

`ActiveActiveShardingReplicaSettings` could become `ReplicaSettings`. 

And `ActiveActiveShardingSettings`, `ReplicatedShardingSettings`

Reply:
Hmm. If we don't have the type params on the class we lock down any use without envelopes and custom extractors and why would we do that here but not for regular Entities? But perhaps we could have a factory method that avoid having to specify it, I 'll look into that and if we can cut those names a bit shorter.

Reply:
Could get rid of one type param on the `ActiveActiveSharding` at least.